internal/controllers/synthesis/gc.go

```diff
@@ -48,6 +48,16 @@ func (p *podGarbageCollector) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, nil
 	}
 	logger = logger.WithValues("podName", pod.Name, "podNamespace", pod.Namespace)
+	
+	// Check for image pull errors early in the reconciliation process
+	if errMsg, hasError := checkImagePullError(pod); hasError {
```

Comment:
This is getting close, just spend a little bit of time refactoring into the style of the project and I think we're good to go.